openedx/core/djangoapps/user_api/views.py

```diff
@@ -168,6 +171,15 @@ class RegistrationView(APIView):
         "terms_of_service",
     ]
 
+    FIELD_TYPE_MAP = {
+        forms.CharField: "text",
+        forms.PasswordInput: "password",
+        forms.ChoiceField: "select",
+        forms.Textarea: "textarea",
+        forms.BooleanField: "checkbox",
+        forms.EmailField: "email",
+    }
```

Comment:
I think you should move this definition into `openedx.core.djangoapps.user_api.helpers.FormDescription` where all the related constants are. Otherwise it could get out of sync. Also that would make it more clear what the meaning is of these magic values.